test/unit/model/search/ResultWatcherTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021  (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoOutcomeUi\unit\model\search;
+
+use core_kernel_classes_Resource;
+use oat\generis\test\TestCase;
+use oat\oatbox\log\LoggerService;
+use oat\oatbox\user\User;
+use oat\oatbox\user\UserService;
+use oat\tao\model\AdvancedSearch\AdvancedSearchChecker;
+use oat\tao\model\search\SearchProxy;
+use oat\tao\model\taskQueue\QueueDispatcherInterface;
+use oat\taoDelivery\model\execution\DeliveryExecutionInterface;
+use oat\taoDelivery\models\classes\execution\event\DeliveryExecutionCreated;
+use oat\taoOutcomeUi\model\search\ResultCustomFieldsService;
+use oat\taoOutcomeUi\model\search\ResultsWatcher;
+use PHPUnit\Framework\MockObject\MockObject;
+
+class ResultWatcherTest extends TestCase
+{
+    /**
+     * @var ResultsWatcher
+     */
+    private $subject;
+    /**
+     * @var AdvancedSearchChecker|MockObject
+     */
+    private $advancedSearchChecker;
+    /**
+     * @var MockObject|string
+     */
+    private $queueDispatcher;
+    /**
+     * @var Search|MockObject
+     */
+    private $searchService;
```

Comment:
PSR + CR agreement + wrong annotation

```suggestion
    /**
     * @var ResultsWatcher
     */
    private $subject;
    
    /**
     * @var AdvancedSearchChecker|MockObject
     */
    private $advancedSearchChecker;
    
    /**
     * @var QueueDispatcherInterface|MockObject
     */
    private $queueDispatcher;
    
    /**
     * @var Search|MockObject
     */
    private $searchService;
```